### DISCUSSION TITLE
Internal Server Error with Radicale version 3.3.0-1 using Python-radicale-decsync 2.1.0-2

### DISCUSSION THREAD

**Discussion by thief-of-times** [Kozea/Radicale#1607] [External Plugins]
I use Radicale as localhost with the Python-radicale-decsync plugin to use my Decsync-calendar in Thunderbird. The calendar data gets syncronized from my smartphone with Syncthing. Since the last update of Radicale and Python in Oktober 16 2024, the Radicale server isn't accessible any more. The login page at localhost:5232 is up, but when I try to log in, I receive the message "Error: 500 Internal Server Error". New calendar events don't arrive in Thunderbird, either. I can't tell if this is an error related to Radicale or Python. 

According to the log (`journalctl --user --unit radicale.service`), the radicale server starts at system boot:
```
Okt 20 20:59:32 device systemd[1987]: Started A simple CalDAV (calendar) and CardDAV (contact) server.
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] Logging of backtrace is disabled in this loglevel
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] Logging of backtrace is disabled in this loglevel
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] Loaded default config
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] Loaded config file '/home/user/.config/radicale/config'
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] Starting Radicale
Okt 20 20:59:33 device radicale[2000]: [2000] [WARNING] No user authentication is selected: '[auth] type=none' (insecure)
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] auth type is 'radicale.auth.none'
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] storage type is 'radicale_storage_decsync'
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] rights type is 'radicale.rights.owner_only'
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] web type is 'radicale.web.internal'
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] hook type is 'radicale.hook.none'
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] permit delete of collection: True
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] permit overwrite of collection: True
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] Listening on '[::1]:5232'
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] Listening on '127.0.0.1:5232'
Okt 20 20:59:33 device radicale[2000]: [2000] [INFO] Radicale server ready
```

When I try to access the calender via browser or Thunderbird, I receive the following messages in the log:
```
Okt 20 21:05:25 device radicale[2000]: [2000/Thread-1 (process_request_thread)] [INFO] PROPFIND request for '/decsync/calendars-a91cf596-17e1-4ef1-950a-99bf67fa62bb/' with depth '0' received from 127.0.0.1 using 'Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Thunderbird/128.3.1'
Okt 20 21:05:25 device radicale[2000]: [2000/Thread-1 (process_request_thread)] [ERROR] An exception occurred during PROPFIND request on '/decsync/calendars-a91cf596-17e1-4ef1-950a-99bf67fa62bb/': Storage.discover() takes from 2 to 4 positional arguments but 5 were given
```

The issue occurred with the following program versions:
Radicale: 3.3.0-1
Python: 3.12.7-1
Python-radicale-decsync: 2.1.0-2

These are my system specifications:
OS: Garuda Linux x86_64
Kernel: 6.11.4-zen1-1-zen
DE: GNOME 47.1
WM: Mutter (Wayland)
CPU: 11th Gen Intel(R) Core(TM) i5-1135G7 (8) @ 4.20 GHz
GPU: Intel Iris Xe Graphics @ 1.30 GHz [Integrated]

**Comment by pbiering** [comment]
Please contact author of 3rd party plugin, so far _Radicale_ has no defined stable API for to be called by 3rd party plugin and internal APIs are subject to be changed at any time. In addition such 3rd party plugins are also not in scope for any tests.

**Comment by thief-of-times** [comment]
Thanks, I did that and reported the issue here: https://github.com/39aldo39/Radicale-DecSync/issues/33. The issue indeed seems to be caused by the plugin because the Radicale server is accessible as lang as I uninstall the plugin and rename the config-file. Am I supposed to close this issue here because it is caused by the plugin?
